docs/mapping/submaps.md

```diff
@@ -0,0 +1,162 @@
+# Guide to Submaps
+
+!!! note
+
+    This guide was originally written by developer Dreamix for Aurora Station, the
+    original codebase the `mapmanip` library was written in. Despite the difference
+    in appearance of the screenshots, the guide is still applicable.
+
+Submaps are a mechanism to allow copy-pasting parts of maps (submaps) into other
+maps, implemented with the `mapmanip` library. `mapmanip` is a library for map
+manipulation, and happens before the map is actually loaded by the server. This
+is implemented using Rust, instead of DM code. Map manipulations are configured
+and defined by adding a `.jsonc` file corresponding to a `.dmm` map. Currently,
+the only available map manipulation is submap extraction/insertion.
+
+## But Why
+
+This allows for adding variations to maps, beyond just using `/obj/random/...`
+items. Variations of bigger parts of maps, like, for example, whole rooms. Or
+even whole layouts, making it so a map looks different every time it is visited.
+
+## How To Do It
+
+1.  Firstly, define two markers for each submap. The types need to be unique, and
+    the types are used later to define where to extract the submap from, and where
+    to insert it. Here, we want to add some variation to the small warehouse storage
+    room.
+
+    ![](./images/mapmanip_types.png)
+
+2.  Figure out the area that the submap will be inserted into on the map, and add
+    the insert marker there. In this case the map is
+    `maps/sccv_horizon/sccv_horizon-1_deck_1.dmm`, and the size of that area is 3x5
+    tiles, being the little side warehouse. Here, we want the submaps to only be the
+    "cargo" or "contents" of the warehouse. So this room is already fully
+    functional, with lights, cameras, pipes, etc. Do note that the "submap edge"
+    helper markers are just visual helpers, and they have no function.
+
+    ![](./images/mapmanip_inplace.png)
+
+3.  Create a new map file that will contain the submap variants to extract from.
+    In this case it is `maps/sccv_horizon/submaps/ops_warehouse_small_storage.dmm`.
+    Add all the possible variants here, each having the same extract marker. They
+    must all be the chosen size, in this case being 3x5 tiles. We want the submaps
+    to only contain "cargo" or "contents" of the warehouse, so both the turf and
+    area are set to "noop". How are "noop" areas and turfs handled, see the next
+    part of this guide.
+
+    ![](./images/mapmanip_contents.png)
+
+4.  Create the mapmanip configuration file, that will actually define the map
+    manipulation, and make it happen. In this case it is
+    `maps/sccv_horizon/sccv_horizon-1_deck_1.jsonc`. Do notice how the path and
```

Comment:
Document with para specific example perhaps

Reply:
Done, I updated some of the paths and screenshots to apply to Boxstation. I haven't replaced all the screenshots though, there's a lot, and it would be painful compared to the benefit.